app.py

```diff
@@ -1,17 +1,286 @@
 import gradio as gr
 import plotly.graph_objects as go
-from datasets import load_dataset
+# import ee
+# # import geemap
+
+# # GEE
+# service_account = 'climatebase-july-2023@ee-geospatialml-aquarry.iam.gserviceaccount.com'
+# credentials = ee.ServiceAccountCredentials(service_account, 'service_account.json')
+# ee.Initialize(credentials)
+
+# # Gradio dataset
+# dataset = load_dataset("gradio/NYC-Airbnb-Open-Data", split="train")
+# df = dataset.to_pandas()
+
+import os
+import duckdb
+import pandas as pd
+import datetime
 import ee
 # import geemap
+import yaml
+
+# Define constants
+MD_SERVICE_TOKEN = 'md_service_token.txt'
+# to-do: set-up with papermill parameters
+DATE='2020-01-01'
+YEAR = 2020
+LOCATION=[-74.653370, 5.845328]
+ROI_RADIUS = 20000
+GEE_SERVICE_ACCOUNT = 'climatebase-july-2023@ee-geospatialml-aquarry.iam.gserviceaccount.com'
+GEE_SERVICE_ACCOUNT_CREDENTIALS_FILE = 'ee_service_account.json'
+INDICES_FILE = 'indices.yaml'
+START_YEAR = 2015
+END_YEAR = 2022
+
+class IndexGenerator:
```

Comment:
We should place `IndexGenerator` in its own file. e.g. `utils/index_generator.py`.